### PR TITLE
Notebook: Escaping code completion pop-up disables cell edit mode

### PR DESCRIPTION
**Pull request by dhuebner**
#### What it does

Fixes the bogus behavior where the cell editor lose focus after closing the code completion widget with `Escape`.
Also adds some Playwright tests. 

See #14327
#### How to test

1. Create a notebook an add `print("Test")` in a code cell.
2. In the next line press `Ctrl+Space` - a code completion widget should pop up.
3. Press `ESC`
4. Completion widget should close and the editor still be in edit mode. Means you can continue typing. Before this change the focus was lost and the user had to press `Enter` to continue typing.  

#### Review checklist

- [x] As an author, I have thoroughly tested my changes and carefully followed [the review guidelines](https://github.com/theia-ide/theia/blob/master/doc/pull-requests.md#requesting-a-review)

#### Reminder for reviewers

- As a reviewer, I agree to behave in accordance with [the review guidelines](https://github.com/theia-ide/theia/blob/master/doc/pull-requests.md#reviewing)

### CODE REVIEW

**Comment by dhuebner** [comment]
@jonah-iden 
I added some more additional tests, now it's ready for re-view.

File: .github/workflows/playwright.yml [hunk 1]
```diff
@@ -67,4 +67,4 @@ jobs:
           path: |
             examples/playwright/test-results/
             examples/playwright/playwright-report/
-          retention-days: 2
+          retention-days: 7
```
Comment (jonah-iden) [inline .github/workflows/playwright.yml:70]: why are we increasing this by so much?

**Review by jonah-iden — APPROVED** [review]
LGTM, not sure about the retention days though.

File: .github/workflows/playwright.yml (hunk 1 above)
Comment (dhuebner) [reply .github/workflows/playwright.yml:70]: Sometimes (sickness/holiday) it takes some time to go back to the failed PR and check the results. In my case 5 days would be probably enough thought.

**Pull request merged**